cpp/source/utils.templates.h

```
@@ -8,12 +8,25 @@
 #include <cap/utils.h>
 #include <boost/lexical_cast.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/regex.hpp>
 #include <boost/foreach.hpp>
 #include <type_traits>
 
 namespace cap
 {
 
+namespace internal
+{
+void filter_boolean(std::string &b)
+{
+  boost::algorithm::to_lower(b);
+  if (b.compare("true") == 0)
+    b = "1";
+  else if (b.compare("false") == 0)
+    b = "0";
```

This should throw an exception in the `else` case.

---

Not necessary, the `boost::lexical_cast<boost>()` will throw otherwhise